### PR TITLE
Build in GH actions and push to Dockerhub head

### PR DESCRIPTION
**Pull request by dragonsahead**
Building master in GH actions and the pushing images to dockerhub

We need to set up DOCKERHUB_TOKEN and DOCKERHUB_USERNAME in the Metabase repository secrets (yes, it's a personal token) to make this work.

As soon as this is set up and merged, we will have to disconnect the webhook of Dockerhub so images are not being overwritten

### CODE REVIEW

File: .github/workflows/build-and-push-head-images.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Build and push head branch container images
```
Comment (ariya) [inline .github/workflows/build-and-push-head-images.yml:1]: Can you make it shorter, e.g. `Dockerhub` or something?
The name will appear on the sidebar of https://github.com/metabase/metabase/actions and if it's too long, it'll be cut off.

File: .github/workflows/build-and-push-head-images.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Create multi-arch image
```
Comment (ariya) [inline .github/workflows/build-and-push-head-images.yml:24]: Just to confirm, without passing `--platform`, is it already default to multi-arch?

File: .github/workflows/build-and-push-head-images.yml (hunk 1 above)
Comment (paoliniluis) [reply .github/workflows/build-and-push-head-images.yml]: yup, done  :)

File: .github/workflows/build-and-push-head-images.yml (hunk 2 above)
Comment (paoliniluis) [reply .github/workflows/build-and-push-head-images.yml]: Sorry, I missed tweaking this text, I just changed to a simple "create container image"

**Review by ariya — APPROVED** [review]
:+1:

**Comment by rlotun** [comment]
repo secrets set up you *should* be good to go @paoliniluis

**Pull request merged**